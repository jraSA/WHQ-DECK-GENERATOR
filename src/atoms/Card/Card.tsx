import React from 'react';
import { Icon } from 'react-native-elements';
import styled from 'styled-components/native';

import themeSchema, { cardSize, cardMargin } from '../../theme/themeSchema';

interface cardPropsI {
    source: any;
    onPress?: () => void | undefined;
    onRevert?: () => void | undefined;
}

const CardContainer = styled.View`
    flex: 1;
    width: ${cardSize.width}px;
    margin: ${cardMargin}px;
    justify-content: center;
    align-items: center;
`;

const IconContaier = styled.View`
    flex: 1;
`;

const TouchableCard = styled.TouchableHighlight`
    width: ${cardSize.width}px;
    height: ${cardSize.height}px;
    border-radius: ${themeSchema.radii[6]}px;
`;

const ImageCard = styled.Image`
    width: ${cardSize.width}px;
    height: ${cardSize.height}px;
    border-radius: ${themeSchema.radii[6]}px;
`;

const Card = (props: cardPropsI) => {
    const { source, onPress, onRevert } = props;

    const renderRevertButton = () => {
        return (
            onRevert && (
                <IconContaier>
                    <Icon
                        name="cached"
                        size={themeSchema.sizes.icon.xsmall}
                        color={themeSchema.colors.blues[700]}
                        onPress={onRevert}
                        reverse
                    />
                </IconContaier>
            )
        );
    };
    return (
        <CardContainer>
            <TouchableCard onPress={onPress}>
                <ImageCard source={source} />
            </TouchableCard>
            {renderRevertButton()}
        </CardContainer>
    );
};

export default Card;
