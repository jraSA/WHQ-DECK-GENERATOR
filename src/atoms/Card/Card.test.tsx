import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from './index';

describe('Card Snapshot test', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Card />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
