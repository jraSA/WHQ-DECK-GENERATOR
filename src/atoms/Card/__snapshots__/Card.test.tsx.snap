// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginBottom": 40,
        "marginLeft": 40,
        "marginRight": 40,
        "marginTop": 40,
        "width": 233,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 22,
          "height": 360,
          "width": 233,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "borderRadius": 22,
            "height": 360,
            "width": 233,
          },
          null,
        ]
      }
    />
  </View>
</View>
`;
